partcad/src/partcad/project_factory_git.py

```
@@ -86,82 +113,113 @@ def _clone_or_update_repo(self, repo_url, cache_dir=None):
-                        if before != after:
-                            pc_logging.info("Updated the GIT repo: %s" % self.import_config_url)
-                        if before != after or user_config.force_update:
-                            with open(guard_path, "w") as f:
-                                if self.import_revision is None:
-                                    f.write(str(after))
-                                else:
-                                    f.write(self.import_revision)
-                except Exception as e:
-                    pc_logging.error("Failed to update a repo: %s" % e)
+                            pc_logging.error("Failed to update repo after %s retries" % attempt)
+                    except Exception as e:
```

Ruff (0.8.2)

Do not catch blind exception: `Exception`

(BLE001)